src/viewer.js

```diff
@@ -594,72 +876,151 @@ class VolumeImageViewer {
-              img.src = window.URL.createObjectURL(blob);
+              const frameData = {
+                frames: renderedFrame,
+              };
+
+              img.src = this[_renderingEngine].createURLFromRGBImage(frameData);
             }
           );
         } else {
-          // TODO: support "image/jp2" and "image/jls"
-          const mediaType = 'image/jpeg';
+          // allowed mediaTypes: http://dicom.nema.org/medical/dicom/current/output/chtml/part18/sect_8.7.4.html  
+          // we use in order: jls, jp2, jpx, jpeg. Finally octet-stream if the first retrieve will fail.
+          const jlsMediaType = 'image/jls';
+          const jlsTransferSyntaxUIDlossless = '1.2.840.10008.1.2.4.80';
+          const jlsTransferSyntaxUID = '1.2.840.10008.1.2.4.81';
+          const jp2MediaType = 'image/jp2';
+          const jp2TransferSyntaxUIDlossless = '1.2.840.10008.1.2.4.90';
+          const jp2TransferSyntaxUID = '1.2.840.10008.1.2.4.91';
+          const jpxMediaType = 'image/jpx';
+          const jpxTransferSyntaxUIDlossless = '1.2.840.10008.1.2.4.92';
+          const jpxTransferSyntaxUID = '1.2.840.10008.1.2.4.93';
+          const jpegMediaType = 'image/jpeg';
+          const jpegTransferSyntaxUID = '1.2.840.10008.1.2.4.50';
+
+          const octetStreamMediaType = 'application/octet-stream';
+          const octetStreamTransferSyntaxUID = '1.2.840.10008.1.2.1';
+
           const retrieveOptions = {
             studyInstanceUID,
             seriesInstanceUID,
             sopInstanceUID,
             frameNumbers,
             mediaTypes: [
-              { mediaType, transferSyntaxUID: '1.2.840.10008.1.2.4.50' }
+              { mediaType : jlsMediaType, transferSyntaxUID : jlsTransferSyntaxUIDlossless },
+              { mediaType : jlsMediaType, transferSyntaxUID : jlsTransferSyntaxUID },
+              { mediaType : jp2MediaType, transferSyntaxUID : jp2TransferSyntaxUIDlossless },
+              { mediaType : jp2MediaType, transferSyntaxUID : jp2TransferSyntaxUID },
+              { mediaType : jpxMediaType, transferSyntaxUID : jpxTransferSyntaxUIDlossless },
+              { mediaType : jpxMediaType, transferSyntaxUID : jpxTransferSyntaxUID },
+              { mediaType : jpegMediaType, transferSyntaxUID : jpegTransferSyntaxUID }
             ]
           };
-          options.client.retrieveInstanceFrames(retrieveOptions).then(
+          this[_options].client.retrieveInstanceFrames(retrieveOptions).then(
             (rawFrames) => {
-              const blob = new Blob(rawFrames, {type: mediaType});
-              img.src = window.URL.createObjectURL(blob);
+              const frameData = {
+                frames: rawFrames[0],
+                bitsAllocated,
+                pixelRepresentation,
+                columns,
+                rows
+              };
+
+              img.src = this[_renderingEngine].createURLFromRGBImage(frameData);
+            }
+          ).catch(
```

Comment:
Currently, the error will still be printed into the console. We may want to improve the error handling of the promise on the dicomweb-client side. Thoughts?

Reply:
![image](https://user-images.githubusercontent.com/7985338/116862079-810ea780-ac04-11eb-862d-fc52e1a43ce0.png)
the last two errors are from dciomweb-client and we could add a flag in the retrieve methods to hide the errors (i.e. just disabling whit console.errors https://github.com/dcmjs-org/dicomweb-client/blob/beeb207ea8589461c43eb8c5e9d43c2f70d92c54/src/api.js#L152-L158). See PR https://github.com/dcmjs-org/dicomweb-client/pull/35

However we will still have this one:
![image](https://user-images.githubusercontent.com/7985338/116862644-5cff9600-ac05-11eb-9353-6c3b677ca8f2.png)
which is from the server.